packages/tools/pluggable-widgets-tools/configs/rollup-plugin-collect-dependencies.js

```diff
@@ -0,0 +1,155 @@
+import { basename, dirname, extname, join } from "path";
+import { copy, readJson, writeJson } from "fs-extra";
+import { promises } from "fs";
+import { rm } from "shelljs";
+
+export function collectDependencies(
+    externals,
+    outputDir,
+    shouldCopyNodeModules = true,
+    widgetName,
+    shouldRemoveNodeModules = true,
+    isJSAction = false
+) {
+    const nativeDependencies = [];
+    const jsActionsDependencies = [];
+    const nodeModulesPath = join(outputDir, "node_modules");
+    return {
+        name: "collect-native-deps",
+        async buildStart() {
+            if (!shouldCopyNodeModules) {
+                return;
+            }
+            if (shouldRemoveNodeModules) {
+                rm("-rf", nodeModulesPath);
+            }
+            nativeDependencies.length = 0;
+        },
+        async resolveId(source) {
+            if (source.startsWith(".")) {
+                return null;
+            }
+            return checkLibraries(
+                externals,
+                source,
+                shouldCopyNodeModules,
+                nativeDependencies,
+                isJSAction,
+                jsActionsDependencies
+            );
+        },
+        async writeBundle() {
+            if (!shouldCopyNodeModules) {
+                return;
+            }
+            await Promise.all(
+                [...nativeDependencies, ...jsActionsDependencies].map(async dependency => {
+                    await copyJsModule(dependency.dir, join(nodeModulesPath, dependency.name));
+                    for (const transitiveDependency of await getTransitiveDependencies(dependency.name, externals)) {
+                        await copyJsModule(
+                            dirname(require.resolve(`${transitiveDependency}/package.json`)),
+                            join(nodeModulesPath, dependency.name, "node_modules", transitiveDependency)
+                        );
+                    }
+                })
+            );
+            await writeNativeDependenciesJson(nativeDependencies, outputDir, widgetName);
+        }
+    };
+}
+
+async function checkLibraries(
```

Comment:
It is not checking a library, maybe `tryResolveNativeDependnecy`? Or alternatively just inline it back...

Reply:
its a recursive function now.. cant just inline :)

Reply:
We renamed to `tryResolveDependency` as this does not contains only native deps check